concrete/src/booleans/server_key.rs

```diff
@@ -0,0 +1,94 @@
+use super::client_key::GenericBoolClientKey;
+use super::parameters::BooleanParameterSet;
+use super::types::GenericBool;
+use concrete_boolean::server_key::{BinaryBooleanGates, ServerKey};
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(doc, cfg(feature = "booleans"))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone)]
+pub struct GenericBoolServerKey<P>
+where
+    P: BooleanParameterSet,
+{
+    pub(in crate::booleans) key: ServerKey,
+    _marker: std::marker::PhantomData<P>,
+}
+
+impl<P> GenericBoolServerKey<P>
+where
+    P: BooleanParameterSet,
+{
+    pub(crate) fn new(key: &GenericBoolClientKey<P>) -> Self {
+        Self {
+            key: ServerKey::new(&key.key),
+            _marker: Default::default(),
+        }
+    }
+
+    pub(in crate::booleans) fn and(
+        &mut self,
+        lhs: &GenericBool<P>,
+        rhs: &GenericBool<P>,
+    ) -> GenericBool<P> {
+        let ciphertext = self.key.and(&lhs.ciphertext, &rhs.ciphertext);
+        GenericBool::<P>::new(ciphertext, lhs.id)
```

Comment:
If the ids are different, then the computation is likely to be wrong as it means the parameters used for the ciphertexts
are likely not the same thus likely not compatible. 

The only case where it can happens is when both lhs and rhs are dynamic types, since in the case the `<P>` type 
is the same (eg: `P = IntegerParameterSet`, so `P::Id = IntegerTypeId` with `pub struct IntegerTypeId(pub usize);`

(
In case of static types, each static type has its own `struct` (`P`) for parameters, which means that 
for example If one tries to `add`  a FheUint2 (`P = FheUint2Parameters, `P::Id  = FheUint2Id`) with a
FheUint3  (`P = FheUint3Parameters, `P::Id  = FheUint3Id`)  since the `P` are different, the program wont compile
).

For now I choose not to add an explicit check (eg: `assert_eq!(lhs.id, rhs.id`), as I though it might be too restrictive
for dynamic types, as the user has full control on the paramters, and my create to different types with different but compatible parameters. 
However not i'm really set on this, and this might change later.